cluster_agent/jobbergate/submit.py

```diff
@@ -86,38 +70,48 @@ async def submit_job_script(
         raise_exc_class=JobSubmissionError,
         do_except=notify_submission_rejected.report_error,
     ):
-
-        email = pending_job_submission.job_submission_owner_email
-        name = pending_job_submission.application_name
+        email = pending_job_submission.owner_email
+        name = pending_job_submission.name
         mapper_class_name = user_mapper.__class__.__name__
         logger.debug(
             f"Fetching username for email {email} with mapper {mapper_class_name}"
         )
         username = await user_mapper.find_username(email)
         logger.debug(f"Using local slurm user {username} for job submission")
 
-        job_script = get_job_script(pending_job_submission)
-
         submit_dir = (
             pending_job_submission.execution_directory
             or SETTINGS.DEFAULT_SLURM_WORK_DIR
         )
 
-        for path, file_content in pending_job_submission.job_script_files.files.items():
-            local_script_path = submit_dir / path
+        job_script = None
+
+        for filename, metadata in pending_job_submission.job_script.files.items():
+            local_script_path = submit_dir / filename
             local_script_path.parent.mkdir(parents=True, exist_ok=True)
-            local_script_path.write_text(file_content)
+
+            response = await backend_client.get(metadata.url)
+            response.raise_for_status()
+            local_script_path.write_bytes(response.content)
+
+            if metadata.file_type == FileType.ENTRYPOINT:
+                job_script = response.content.decode("utf-8")
```

Comment:
All files are written to the disk in the submission directory, including the job script itself (entry point) and supporting files. Job script content is also kept in memory because it is sent to slurm rest API. This reproduces the previous behavior.

I'll make a note to extract it as a function to make it more readable.

Reply:
if it is going to be submitted by slurmrestd, why are the files needed to be written in disk?

Reply:
Support files are expected there by the job script itself or any other process started by it. The job script is really not needed on the disk, but it is a feature requested by the users.